first draft to merge the list of machines in a page

sblondon opened the pull request:
This pull request add 2 urls usables to display data about the machines:
/machines/
/machines/name
They are mapped on 2 differents views and templates.

The code is minimal. I needed to change the LDAP protocol from ldaps to ldap. With ldaps, the request were refused. This modification is not included into the patch.

The html code does not include the mandatory code to be valid (html, body tags). I didn't add it to simplify this code review. I plan to add it later.

After writing it, I think it would be better to include this code and templates in a specific directory (like 'web' or 'www') to separate it from the rest of the application. What do you think about it?

What do you think about the code of the pull/request too?

sblondon committed:
basic files for webrequest about machines

sblondon committed:
fill templates with informations about machines from the LDAP

sblondon committed:
cleaning: remove a print statement

pabs3 commented:
I prefer these URLs instead:

/machines
/machine/draghi

Please merge the cleaning commit into the first commit.

It would be nice if the templates matched what we have already.

ldaps is supported by db.debian.org, not sure why you got an error.

I don't know django very well, but ISTR that it has a way to map views
to URLs in templates so you don't have to hard-code URLs in hrefs.

sblondon committed:
accept machine names with dot or minus characters

sblondon committed:
map views to URLs in templates to avoid hardcoded URLs

sblondon commented:
- the last commit is rebased (fixed up)
- mapping views -> url in templates: done
- add the rest of the current template: todo (I plan to do it the week.)

Are you sure about the '/machine/name' instead of the plural version ('/machines/name')?
I see more example for the latest one (for example http://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api or http://www.restapitutorial.com/lessons/restfulresourcenaming.html). 
The django tutorial provide another way to do it (/machines and /machines/machine/name).
If you prefer your way, I will do it like you prefer.

pabs3 commented:
On Tue, 2016-05-24 at 17:42 -0700, sblondon wrote:

> Are you sure about the '/machine/name' instead of the plural version
> ('/machines/name')?

I prefer it but either way works.

> I see more example for the latest one (for example
> http://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api
> or http://www.restapitutorial.com/lessons/restfulresourcenaming.html)

Those are about REST APIs, which is not what this patch is about.

> The django tutorial provide another way to do it (/machines and
> /machines/machine/name).

That is even more ugly :)

-- 
bye,
pabs

http://bonedaddy.net/pabs3/

sblondon committed:
url for detailed page about a server is /machine/hostname

sblondon committed:
use a base template for the templates of servers list and detailled info about a server

sblondon committed:
the main structure of the pages render like the current production site

sblondon committed:
the details and list of server templates match the style of the current production site

sblondon committed:
add hostname, processor and ssh key to the server detail page

sblondon commented:
In the new commits : 
- the detail URL is /machine/name. 
- the structure of the pages are the same as in production. The static files (js and css) are directly linked to the production version. I plan to copy them into a media/ directory:
-- media/vendors/ for jquery and other libraries
-- at the root of media/ the files specific to the website (like debian.css for example)

Do you agree with it?

It seems that some data are not available in common.models.DebianHost (or __LdapDebianServer) so I can't insert them into the pages. Did I missed something?

pabs3 commented:
On Wed, 2016-05-25 at 16:35 -0700, sblondon wrote:

> the detail URL is /machine/name.

Thanks.

> I plan to copy them into a media/ directory: -- media/vendors/ for
> jquery and other libraries

Please don't create embedded code copies, we will have ud depend on the
Debian packages of any external libraries needed, as we do for the
existing site that uses userdir-ldap-cgi.

https://wiki.debian.org/EmbeddedCodeCopies
https://anonscm.debian.org/cgit/mirror/userdir-ldap-cgi.git/

> It seems that some data are not available in common.models.DebianHost
> (or __LdapDebianServer) so I can't insert them into the pages. Did I
> missed something?

Which data are you talking about? I guess you will need to add the data
to those classes.

Some more comments:

These files contain trailing whitespace:

src/templates/machines/all.html
src/templates/machines/details.html

I would use os.path.join() instead of + "/ 

The way the urlpatterns stuff is looks weird to me, I especially wonder
if it works when passing parameters.

ROOT_URLCONF shouldn't have two definitions.

I would call all_machines list instead.

-- 
bye,
pabs

https://wiki.debian.org/PaulWise

sblondon committed:
refactoring: os.path.join instead of concatenation; remove redefiniton of a constant

sblondon committed:
add links file to reuse jquery libraries installed on the system

sblondon committed:
display url as link for details machine page

sblondon committed:
page with all machines uses wiki_link

sblondon committed:
text to display link instead of the URL when it's available

sblondon committed:
manage several wiki links in the same string

sblondon committed:
remove trailing spaces and space before ':' characters in template files

sblondon commented:
There are still a lot of stuff to do but there are some news:

> Please don't create embedded code copies, we will have ud depend on the
> Debian packages of any external libraries needed, as we do for the
> existing site that uses userdir-ldap-cgi.
>
> https://wiki.debian.org/EmbeddedCodeCopies
> https://anonscm.debian.org/cgit/mirror/userdir-ldap-cgi.git/

I found the debian/links file for userdir-ldap-cgi and copy-pasted it with 'ud' as name. It probably not sufficient but it's a start.


> > It seems that some data are not available in common.models.DebianHost
> > (or __LdapDebianServer) so I can't insert them into the pages. Did I
> > missed something?

> Which data are you talking about? I guess you will need to add the data
> to those classes.

Yes, it was my conclusion. It' probably my next task, competing with understanding why ldaps protocol raises an error (and not ldap).


> These files contain trailing whitespace:
> 
> src/templates/machines/all.html
> src/templates/machines/details.html

Fixed.


> I would use os.path.join() instead of + "/ 

Fixed.


> The way the urlpatterns stuff is looks weird to me, I especially wonder
> if it works when passing parameters.

Yes, it looks weird due to the difference from '/machines' and '/machine/name'. But perhaps you talk about something else?


> ROOT_URLCONF shouldn't have two definitions.

Fixed.


> I would call all_machines list instead.

Do you talk about the file src/templates/machines/all.html?


I added transformation from string like '[[gluck.debian.org]]' to '<ahref="gluck.debian.org">gluck.debian.org</a>' too.

sblondon committed:
display more details about machines (description, distribution, bandwidth and status)

sblondon committed:
display more details about machines (admin contact, memory and disk)

sblondon committed:
remove hostnames info because it's not displayed in the equivalent page at db.debian.org

sblondon committed:
add sponsors info

sblondon committed:
add access info on detail page

sblondon committed:
hide empty fields

sblondon committed:
adding description, sponsors, status and access of all machines

sblondon committed:
add tests for wiki_link filter and add quote to URL by wiki_links

sblondon committed:
add http protocol to the links created by wiki_links

sblondon committed:
wiki_links pareses link with parenthesis in description

sblondon committed:
wiki_links pareses link with ampersand in description

sblondon commented:
Is it ok to use javascript for sorting the table of all machines or should I do it server-side?

sblondon committed:
wiki_links parses deeplink

sblondon committed:
wiki_links parses url with description containing characters with accent

sblondon committed:
wiki_links parses url with description containing comas

sblondon commented:
Additionals info about the machines are added.


It seems the LDAP is not available under TLS layer:

stephane@foehn:~/src/ud/src$ telnet db.debian.org 389
Trying 82.195.75.106...
Connected to db.debian.org.
Escape character is '^]'.
^C)^[)
Connection closed by foreign host.

stephane@foehn:~/src/ud/src$ telnet db.debian.org 686
Trying 82.195.75.106...
Trying 2001:41b8:202:deb:1a1a:0:52c3:4b6a...
telnet: Unable to connect to remote host: Network is unreachable

sblondon committed:
remove a trailing space
